### ISSUE TITLE
official YOLO architecture diagram for research reference

### ISSUE THREAD

**Issue by banu4prasad** [ultralytics/ultralytics#22633]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have been training models using YOLOv8, .... YOLO11 on diverse datasets for research purposes. While working on my experiments and preparing my research paper, I noticed that there is no official architecture diagram available for these YOLO versions in the documentation or repository.

Most of the images available online are generic CNN block diagrams or outdated schematics (up to YOLOv5), which don’t accurately represent the modern Ultralytics YOLO architecture — including its backbone, neck (PAN/FPN), and detection head.

It would be extremely helpful if Ultralytics could provide:

- Official architecture diagrams (for YOLOv8, YOLOv9, YOLOv10, and YOLOv11) showing key components such as the backbone (C2f modules, CBS, etc.), neck, and detection layers.

- Or alternatively, a verified schematic or reference illustrating the overall network flow (input → backbone → neck → detection head) for citation in academic work.

### Additional

- Use case: Academic research and model comparison (YOLOv8–YOLOv11)

- Docs searched: https://docs.ultralytics.com/

**Comment by UltralyticsAssistant** [comment]
👋 Hello @banu4prasad, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your request and assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

For your architecture diagram request 📐:
-Please confirm which versions you specifically need diagrams for (YOLO8, YOLO9, YOLO10, YOLO11), and whether you need model-size-specific variants (n/s/m/l/x).  
-If you have preferred detail level (high-level schematic vs. module-level like C2f/CBS/PAN/FPN/head), share that too so we can route this appropriately.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the thoughtful request—there isn’t a single canonical block diagram for each variant yet. The authoritative definitions are the model YAMLs (e.g., [YOLOv8 YAML](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/v8/yolov8.yaml), [YOLOv10n YAML](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/v10/yolov10n.yaml), [YOLO11 YAML](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/11/yolo11.yaml)), and you can cite the model docs: [YOLOv8 docs](https://docs.ultralytics.com/models/yolov8/), [YOLOv9 docs](https://docs.ultralytics.com/models/yolov9/), [YOLOv10 docs](https://docs.ultralytics.com/models/yolov10/), [YOLO11 docs](https://docs.ultralytics.com/models/yolo11/) as well as the [Object Detection Architectures overview](https://www.ultralytics.com/glossary/object-detection-architectures). For a publishable schematic, export and visualize with Netron: 

```bash
yolo export model=yolo11n.pt format=onnx simplify=True dynamic=False
```

We’ll keep this open as a docs enhancement; contributions adding clean SVG diagrams to the model pages are welcome.

**Comment by banu4prasad** [comment]
yeah i tried netron but it will be specific, but for my use case i need a general YOLO diagram

**Comment by glenn-jocher** [comment]
Got it—here’s a concise, model-agnostic YOLO schematic you can cite; it abstracts backbone → neck → head and accommodates NMS-free decoding used by newer variants:

```mermaid
flowchart LR
  I[Input image / video] --> B[Backbone<br/>(e.g., C2f/CSP, CBS)]
  B --> N[Neck<br/>(FPN/PAN multi-scale fusion)]
  N --> H[Decoupled Detection Head]
  H --> P1[Box / class logits]
  H --> P2[Optional heads: mask / pose / OBB]
  P1 --> D[NMS or NMS‑free decoder]
  P2 --> D
  D --> O[Final detections]
```

For terminology and context you can reference the architecture notes in the [YOLOv8 vs YOLO11 comparison](https://docs.ultralytics.com/compare/yolov8-vs-yolo11/) and the YOLOv10 NMS‑free discussion in [YOLO model comparison: YOLO11 vs previous](https://www.ultralytics.com/blog/comparing-ultralytics-yolo11-vs-previous-yolo-models).

**Comment by banu4prasad** [comment]
```mermaid
flowchart LR
  I["Input image / video"] --> B["Backbone\n(e.g., C2f/CSP, CBS)"]
  B --> N["Neck\n(FPN/PAN multi-scale fusion)"]
  N --> H["Decoupled Detection Head"]
  H --> P1["Box / class logits"]
  H --> P2["Optional heads:\nmask / pose / OBB"]
  P1 --> D["NMS or NMS-free decoder"]
  P2 --> D
  D --> O["Final detections"]

**Comment by banu4prasad** [comment]
can i use this diagram?

![Image](https://github.com/user-attachments/assets/ce6522ce-84ba-4f27-a7e9-57836bc4dfe4)

**Comment by glenn-jocher** [comment]
Yes, you can use that diagram as a generic Ultralytics YOLO architecture illustration—just label it as a conceptual overview (not an exact layer-by-layer spec) and cite the Ultralytics docs, for example the [YOLOv5 model page](https://docs.ultralytics.com/models/yolov5/), in your paper.

**Comment by Petros626** [comment]
I tried my luck and created YOLOv11 OBB, hopefully correct, but with the `.pptx` file, you can always suggest changes, beautification etc. I don't got the time to take care of changes, so it's up to the community.

<img width="3250" height="3402" alt="Image" src="https://github.com/user-attachments/assets/650220ba-6c13-42b3-887e-d900b3e46f5e" />

[YOLOv11_OBB_architecture.pptx](https://github.com/user-attachments/files/24911792/YOLOv11_OBB_architecture.pptx)

**Comment by glenn-jocher** [comment]
Thanks for sharing this—if you’d like it to become an “official” reference diagram, please open a PR against the [Ultralytics Docs repository](https://github.com/ultralytics/docs) adding an exported `*.svg` (PPTX source is great too) to the [OBB task docs](https://docs.ultralytics.com/tasks/obb/); a quick way to sanity-check block ordering is to build from `yolo11n-obb.yaml` and compare against the printed layer list via [`Model.info()`](https://docs.ultralytics.com/reference/engine/model/):

```python
from ultralytics import YOLO

m = YOLO("yolo11n-obb.yaml")
m.info(detailed=True)
```
